src/base/system.cpp

```diff
@@ -842,7 +842,14 @@ void *thread_init(void (*threadfunc)(void *), void *u, const char *name)
 #elif defined(CONF_FAMILY_WINDOWS)
 	HANDLE thread = CreateThread(nullptr, 0, thread_run, data, 0, nullptr);
 	dbg_assert(thread != nullptr, "CreateThread failure");
-	// TODO: Set thread name using SetThreadDescription (would require minimum Windows 10 version 1607)
+	HMODULE handle;
+	if (GetModuleHandleExW(GET_MODULE_HANDLE_EX_FLAG_PIN, L"KernelBase.dll", &handle) && handle)
+	{
+		using SetThreadDescriptionFunc = HRESULT(WINAPI *)(HANDLE, PCWSTR);
+		auto pSetThreadDescription = reinterpret_cast<SetThreadDescriptionFunc>(GetProcAddress(handle, "SetThreadDescription"));
+		if (pSetThreadDescription)
+			pSetThreadDescription(thread, windows_utf8_to_wide(name).c_str());
```

Comment:
Using `GetProcAddress` requires suppressing `warning: cast between incompatible function types from 'FARPROC' {aka 'long long int (*)()'} to 'HRESULT (*)(HANDLE, PCWSTR)' {aka 'long int (*)(void*, const wchar_t*)'} [-Wcast-function-type]`:

```suggestion
#ifdef __MINGW32__
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wcast-function-type"
#endif
		auto set_thread_description_function = reinterpret_cast<HRESULT(WINAPI *)(HANDLE, PCWSTR)>(GetProcAddress(kernel_base_handle, "SetThreadDescription"));
#ifdef __MINGW32__
#pragma GCC diagnostic pop
#endif
		if(set_thread_description_function)
			set_thread_description_function(thread, windows_utf8_to_wide(name).c_str());
```